EDID Processing API and proofreading (#37)
AMWA-TV/nmos-edid-connection-management: docs/3.0. Behaviour.md

### Senders

`/media-profiles` allows to pass an array of Media Profiles to a Sender and let a third-party which controls it choose what exact format to use. Also an API user can pass a single Media Profile to specify exact format strictly.

The initial state of Media Profiles of any Sender is empty. Creating a connection with such Sender via IS-05 doesn't involve NMOS EDID Connection Management.

GET request returns the last successfully applied Media Profiles array.

PUT request MUST be validated and if the Device fully supports the proposed media parameters, it initiates an attempt to reconfigure Device and update according Flow and/or Source and/or Sender values. If success, the GET operation MUST start returning the accepted Media Profiles array, otherwise an error MUST be returned and nothing changed.

DELETE request MUST clear the last successfully applied Media Profiles array and NMOS EDID Connection Management MUST fall out of use and return to its initial state.

> ### Receivers
> 
> `/sinks` lists IDs of Sinks associated with a specific Receiver.
> 
> ### Sinks
> 
> `/sinks/{sinkId}` provides Sink resource for an API user. A Sink consists of Resource Core and EDID parsed into JSON.

Reviewer: Slight modifications, the URLs looked like the same at first glance

```suggestion
### Receivers

`/receivers/{receiverId}/sinks` list of IDs corresponding to Sinks associated to the specific Receiver.

### Sinks

`/sinks/{sinkId}` provides the Sink resource which consists of Resource Core and EDID parsed into JSON.
```

Author: I like this change, but I have feeling that a verb is now missed in the 1st sentence.

Author: I would prefer this.

```suggestion
### Receivers

`/receivers/{receiverId}/sinks` returns a list of IDs corresponding to Sinks associated to the specific Receiver.

### Sinks

`/sinks/{sinkId}` provides the Sink resource which consists of Resource Core and EDID parsed into JSON.
```

Reviewer: I like it!

## Senders

> `/senders` returns all Senders on the Node which is associated with the API.
> 

Reviewer: I think the Senders and Receivers should both have one sentence saying "these GUIDs and resource correspond with what is exposed with IS-04". It's the same for both.

```suggestion
```

Author: @prince-chrismc, do you mean in Behaviour or it's enough if only in RAML?

Reviewer: Scanning quickly I did not see this in the RAML, regardless I think the Behavior section here is enough

Author: I fixed it in f78576af69c869e5cf06f64c284c63a55ab5ecb8.

`/media-profiles` allows to pass an array of Media Profiles to a Sender and let a third-party which controls it choose what exact format to use. Also an API user can pass a single Media Profile to specify exact format strictly.

The initial state of Media Profiles of any Sender is empty. Creating a connection with such Sender via IS-05 doesn't involve NMOS EDID Connection Management.

GET request returns the last successfully applied Media Profiles array.

PUT request MUST be validated and if the Device fully supports the proposed media parameters, it initiates an attempt to reconfigure Device and update according Flow and/or Source and/or Sender values. If success, the GET operation MUST start returning the accepted Media Profiles array, otherwise an error MUST be returned and nothing changed.

> DELETE request to MUST clear the last successfully applied Media Profiles array and NMOS EDID Connection Management MUST fall out of use.

Reviewer: Does this make sense?

```suggestion
DELETE request MUST clear the last successfully applied Media Profiles array and NMOS EDID Connection Management MUST fall out of use and return to its initial state.
```
